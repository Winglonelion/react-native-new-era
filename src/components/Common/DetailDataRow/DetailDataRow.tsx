import React, { memo } from 'react';
import {
  Image,
  StyleSheet,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import ContentText from 'components/Base/Text/ContentText';
import Colors from 'theme/colors';
import { PressEvent } from 'types/events';

export interface PropTypes {
  title: string;
  content?: string | number | null;
  onPress?: PressEvent;
  strongChevron?: boolean;
}

const DetailDataRow: React.FC<PropTypes> = memo(
  ({ title, content, onPress, strongChevron = false }) => {
    return (
      <TouchableWithoutFeedback disabled={!onPress} onPress={onPress}>
        <View style={styles.container}>
          <View style={styles.content}>
            <ContentText size={16} style={styles.text}>
              {title}
            </ContentText>
            {content !== null && (
              <ContentText size={16} style={[styles.text, styles.contentTxt]}>
                {content || 'â€”'}
              </ContentText>
            )}
          </View>
          {!!onPress && (
            <Image
              source={
                strongChevron
                  ? require('images/chevron_right_brown.png')
                  : require('images/chevron_right_light_brown.png')
              }
              style={styles.icon}
            />
          )}
        </View>
      </TouchableWithoutFeedback>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    paddingHorizontal: 15,
    alignItems: 'center',
  },
  content: {
    paddingVertical: 15,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    flex: 1,
  },
  contentTxt: {
    color: Colors.lightBrown,
  },
  text: {
    lineHeight: 30,
  },
  icon: {
    width: 24,
    height: 24,
    resizeMode: 'contain',
    marginLeft: 8,
  },
});

export default DetailDataRow;
