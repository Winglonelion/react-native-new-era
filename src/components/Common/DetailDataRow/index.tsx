export { default as default } from './DetailDataRow';
