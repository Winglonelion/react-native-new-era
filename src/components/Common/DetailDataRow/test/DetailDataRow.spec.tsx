import React from 'react';
import { shallow } from 'enzyme';
import DetailDataRow, { PropTypes } from '../DetailDataRow';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<DetailDataRow {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('DetailDataRow check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      title: 'test menu',
      content: 'test data',
      onPress: jest.fn(),
      strongChevron: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props and loading', () => {
    const { wrapper } = setup({
      title: 'test menu',
      strongChevron: false,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot without not strong chevron', () => {
    const { wrapper } = setup({
      title: 'test menu',
      onPress: jest.fn(),
    });
    expect(wrapper).toMatchSnapshot();
  });
});
