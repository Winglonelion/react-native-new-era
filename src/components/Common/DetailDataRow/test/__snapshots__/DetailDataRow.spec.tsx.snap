// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailDataRow check snapshot should match snapshot with props 1`] = `
<TouchableWithoutFeedback
  disabled={false}
  onPress={[MockFunction]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 15,
        }
      }
    >
      <Memo()
        size={16}
        style={
          Object {
            "lineHeight": 30,
          }
        }
      >
        test menu
      </Memo()>
      <Memo()
        size={16}
        style={
          Array [
            Object {
              "lineHeight": 30,
            },
            Object {
              "color": "#9C9079",
            },
          ]
        }
      >
        test data
      </Memo()>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/images/chevron_right_brown.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginLeft": 8,
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
  </View>
</TouchableWithoutFeedback>
`;

exports[`DetailDataRow check snapshot should match snapshot without not strong chevron 1`] = `
<TouchableWithoutFeedback
  disabled={false}
  onPress={[MockFunction]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 15,
        }
      }
    >
      <Memo()
        size={16}
        style={
          Object {
            "lineHeight": 30,
          }
        }
      >
        test menu
      </Memo()>
      <Memo()
        size={16}
        style={
          Array [
            Object {
              "lineHeight": 30,
            },
            Object {
              "color": "#9C9079",
            },
          ]
        }
      >
        —
      </Memo()>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/images/chevron_right_light_brown.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginLeft": 8,
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
  </View>
</TouchableWithoutFeedback>
`;

exports[`DetailDataRow check snapshot should match snapshot without props and loading 1`] = `
<TouchableWithoutFeedback
  disabled={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 15,
        }
      }
    >
      <Memo()
        size={16}
        style={
          Object {
            "lineHeight": 30,
          }
        }
      >
        test menu
      </Memo()>
      <Memo()
        size={16}
        style={
          Array [
            Object {
              "lineHeight": 30,
            },
            Object {
              "color": "#9C9079",
            },
          ]
        }
      >
        —
      </Memo()>
    </View>
  </View>
</TouchableWithoutFeedback>
`;
